conformalInference/R/multisplit.R

```diff
@@ -0,0 +1,193 @@
+#'   Y given features X. Its input arguments should be x: matrix of features,
+#'   and y: vector of responses.
+#' @param predict.fun A function to perform prediction for the (mean of the)
+#'   responses at new feature values. Its input arguments should be out: output
+#'   produced by train.fun, and newx: feature values at which we want to make
+#'   predictions.
+#' @param alpha Miscoverage level for the prediction intervals, i.e., intervals
+#'   with coverage 1-alpha are formed. Default for alpha is 0.1.
+#' @param rho It is a vector of split proportions of length B. Default is NULL.
+#' @param w Weights, in the case of covariate shift. This should be a vector of
+#'   length n+n0, giving the weights (i.e., ratio of test to training feature
+#'   densities), at each of n+n0 the training and test points. Default is NULL,
+#'   which means that we take all weights to be 1.
+#' @param mad.train.fun A function to perform training on the absolute residuals
+#'   i.e., to produce an estimator of E(R|X) where R is the absolute residual
+#'   R = |Y - m(X)|, and m denotes the estimator produced by train.fun.
+#'   This is used to scale the conformal score, to produce a prediction interval
+#'   with varying local width. The input arguments to mad.train.fun should be
+#'   x: matrix of features, and y: vector of absolute residuals. The default for
+#'   mad.train.fun is NULL, which means that no training is done on the absolute
+#'   residuals, and the usual (unscaled) conformal score is used. Note that if
+#'   mad.train.fun is non-NULL, then so must be mad.predict.fun (see next).
+#' @param mad.predict.fun A function to perform prediction for the (mean of the)
+#'   absolute residuals at new feature values. Its input arguments should be
+#'   out: output produced by mad.train.fun, and newx: feature values at which we
+#'   want to make predictions. The default for mad.predict.fun is NULL, which
+#'   means that no local scaling is done for the conformal score, i.e., the
+#'   usual (unscaled) conformal score is used.
+#' @param split Indices that define the data-split to be used (i.e., the indices
+#'   define the first half of the data-split, on which the model is trained).
+#'   Default is NULL, in which case the split is chosen randomly.
+#' @param seed Integer to be passed to set.seed before defining the random
+#'   data-split to be used. Default is NULL, which effectively sets no seed.
+#'   If both split and seed are passed, the former takes priority and the latter
+#'   is ignored.
+#' @param verbose Should intermediate progress be printed out? Default is FALSE.
+#' @param B number of replications. Default is 50.
+#' @param lambda Smoothing parameter. Default is 0.
+#' @param tau It is a smoothing parameter, whose value affects the behavior of the
+#'  function joining the B intervals:
+#'     tau=1-1/B  Bonferroni intersection method
+#'     tau=0 unadjusted intersection
+#' Default is 1-(B+1)/(2*B).
```

Comment:
I think `verbose` should always come last. 

Beyond that, some questions come to mind:

- Are `B`, `lambda`, `tau` the only parameters that differ here than in `conformal.pred.split()`? 
- Is the behavior here going to be identical to the behavior in `conformal.pred.split()` when we set `B=1`?

It would be good to address questions like these (and any other differences to the `conformal.pred.split()` function that you can think of) in the details section of the documentation.